ledger/caching/src/main/scala/com/daml/caching/MappedCache.scala

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.caching
+
+private[caching] final class MappedCache[Key, Value, NewValue](
+    from: Value => NewValue,
+    to: NewValue => Option[Value],
```

Comment:
Here it may be worth noting that in case the mapping returns `None` the value won't be written.